src/CacheManager.SystemRuntimeCaching/MemoryCacheHandle`1.cs

```diff
@@ -70,6 +73,12 @@ public MemoryCacheHandle(ICacheManagerConfiguration managerConfiguration, CacheH
         public override int Count => (int)_cache.GetCount();
 
         /// <inheritdoc />
+        protected override IEnumerable<string> AllKeys()
+        {
+            return _cache.Select(c => c.Key);
```

Comment:
So this would either override Keys() or trim the instance key off before passing the keys up in AllKeys()